MarkBind/markbind
# Allocate space for scrollbar in sitenav and pagenav

**What is the purpose of this pull request?**




**Overview of changes:**
#2217 
Changed the property value of `overflow-y` of the `.nav-component` to `scroll` to allocate space to the scrollbar as discussed [here](https://github.com/MarkBind/markbind/pull/2204#issuecomment-1469873352) so that its appearance and disappearance doesn't cause the icon in the nav bar to shift. 

Since this change is to `.nav-component`, **both the sitenav and pagenav is affected**.

Since the change is to the `main.css` file of the default template, all the other `main.css` files also have to be updated. Currently there isn't a way to automatically update the file after a MarkBind upgrade (#1961), so if this change is accepted, there has to be release note for current users to update their `main.css` file.

**Anything you'd like to highlight/discuss:**
`overflow-y: overlay` is a legacy value alias for `overflow-y: auto` according to the mdn docs [here](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-y#:~:text=Note%3A%20The%20keyword%20value%20overlay%20is%20a%20legacy%20value%20alias%20for%20auto.%20With%20overlay%2C%20the%20scroll%20bars%20are%20drawn%20on%20top%20of%20the%20content%20instead%20of%20taking%20up%20space.), which unfortunately still causes the moving of the content from what I can see. 

The overlay value of the overflow CSS property overall is being deprecated and standardised as [the scrollbar-gutter property](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter), but I see that its [not really supported on Safari browser](https://caniuse.com/mdn-css_properties_scrollbar-gutter). If anyone wants me to look into using this property instead, or any other alternatives, I'll be happy to try it out. 

Previous behaviour:

https://github.com/MarkBind/markbind/assets/105090139/3e97f9fa-3d0c-4bb6-adb8-6450bcabe48f


Current behaviour:

https://github.com/MarkBind/markbind/assets/105090139/2e83f9a2-2440-4136-97e4-be21ef0cf0c7




**Testing instructions:**
Serve the docs using `markbind serve -d`, collapse all sections of the siteNav, open the sections one by one until the scrollbar appears. Ensure that when the scrollbar appears, the icons doesnt shift.



**Proposed commit message: (wrap lines at 72 characters)**
Allocate space for scrollbar in nav components



---

**Checklist:** :ballot_box_with_check:






---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).
